execution_engine_testing/tests/src/test/chainspec_registry.rs

```diff
@@ -0,0 +1,243 @@
+    let genesis_account_hash = Digest::hash(rng.gen::<[u8; 32]>());
+
+    let mut chainspec_registry = ChainspecRegistry::new();
+    chainspec_registry.insert(CHAINSPEC_RAW.to_string(), chainspec_bytes_hash);
+    chainspec_registry.insert(GENESIS_ACCOUNTS_RAW.to_string(), genesis_account_hash);
+
+    let run_genesis_request = RunGenesisRequest::new(
+        *DEFAULT_GENESIS_CONFIG_HASH,
+        *DEFAULT_PROTOCOL_VERSION,
+        DEFAULT_EXEC_CONFIG.clone(),
+        chainspec_registry.clone(),
+    );
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&run_genesis_request);
+
+    let queried_registry = builder
+        .query(None, Key::ChainspecRegistry, &[])
+        .expect("must have entry under Key::ChainspecRegistry")
+        .as_cl_value()
+        .expect("must have underlying cl_value")
+        .to_owned()
+        .into_t::<ChainspecRegistry>()
+        .expect("must convert to chainspec registry");
+
+    let queried_chainspec_hash = queried_registry
+        .get(CHAINSPEC_RAW)
+        .expect("must have entry for chainspec_hash");
+
+    assert_eq!(*queried_chainspec_hash, chainspec_bytes_hash);
+
+    let queried_accounts_hash = queried_registry
+        .get(GENESIS_ACCOUNTS_RAW)
+        .expect("must have entry for genesis accounts");
+
+    assert_eq!(*queried_accounts_hash, genesis_account_hash);
+}
+
+#[ignore]
+#[test]
+#[should_panic]
+fn should_fail_to_commit_genesis_when_missing_chainspec_hash() {
+    let incomplete_chainspec_registry = ChainspecRegistry::new();
+
+    let run_genesis_request = RunGenesisRequest::new(
+        *DEFAULT_GENESIS_CONFIG_HASH,
+        *DEFAULT_PROTOCOL_VERSION,
+        DEFAULT_EXEC_CONFIG.clone(),
+        incomplete_chainspec_registry,
+    );
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&run_genesis_request);
+}
+
+#[ignore]
+#[test]
+#[should_panic]
+fn should_fail_to_commit_genesis_when_missing_genesis_accounts_hash() {
+    let mut rng = rand::thread_rng();
+    let chainspec_bytes_hash = Digest::hash(rng.gen::<[u8; 32]>());
+
+    let mut incomplete_chainspec_registry = ChainspecRegistry::new();
+    incomplete_chainspec_registry.insert(CHAINSPEC_RAW.to_string(), chainspec_bytes_hash);
+
+    let run_genesis_request = RunGenesisRequest::new(
+        *DEFAULT_GENESIS_CONFIG_HASH,
+        *DEFAULT_PROTOCOL_VERSION,
+        DEFAULT_EXEC_CONFIG.clone(),
+        incomplete_chainspec_registry.clone(),
+    );
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&run_genesis_request);
+}
+
+#[ignore]
+#[test]
+fn should_write_chainspec_registry_during_an_upgrade() {
+    let mut rng = rand::thread_rng();
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&DEFAULT_RUN_GENESIS_REQUEST);
+
+    let mut upgraded_chainspec_registry = ChainspecRegistry::new();
+    let chainspec_bytes_hash = Digest::hash(rng.gen::<[u8; 32]>());
```

Comment:
Consider renaming to `upgraded_chainspec_bytes_hash` - it'll make it easier to see what's going on down below at the assert.

Reply:
Applied suggestion